sphinx/conf.py

```diff
@@ -179,3 +189,9 @@
 ]
 # How to display URL addresses: 'footnote', 'no', or 'inline'.
 # texinfo_show_urls = 'footnote'
+
+
+def setup(app):
```

Comment:
I suspect there is some issue with this function or its usage, since when I initially ran ``make html`` (after a ``make clean``) with a build that "succeeded" (strictly when I first ran it once it did not return some ``Extension error`` from the dependencies I was missing & installed in turn) I saw 6 warnings relating to unfound ``_static`` directories, which I checked are in fact there under the relevant sub-dirs of ``cylc/sphinx_ext/`` in the ``cylc-sphinx-extensions`` repo (e.g. as shown [here for ``diff_selection``](https://github.com/cylc/cylc-sphinx-extensions/tree/master/cylc/sphinx_ext/diff_selection) WRT the first warning):

```console
$ make html
sphinx-build -b html -d ../doc/doctrees   . ../doc/html
Running Sphinx v2.0.1

...

copying static files... WARNING: html_static_path entry '/net/home/h06/sbarth/miniconda3/lib/python3.7/site-packages/cylc/sphinx_ext/diff_selection/_static' does not exist
WARNING: html_static_path entry '/net/home/h06/sbarth/miniconda3/lib/python3.7/site-packages/cylc/sphinx_ext/grid_table/_static' does not exist
WARNING: html_static_path entry '/net/home/h06/sbarth/miniconda3/lib/python3.7/site-packages/cylc/sphinx_ext/hieroglyph_theme_addons/_static' does not exist
WARNING: html_static_path entry '/net/home/h06/sbarth/miniconda3/lib/python3.7/site-packages/cylc/sphinx_ext/minicylc/_static' does not exist
WARNING: html_static_path entry '/net/home/h06/sbarth/miniconda3/lib/python3.7/site-packages/cylc/sphinx_ext/practical/_static' does not exist
WARNING: html_static_path entry '/net/home/h06/sbarth/miniconda3/lib/python3.7/site-packages/cylc/sphinx_ext/rtd_theme_addons/_static' does not exist
done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 6 warnings.

The HTML pages are in ../doc/html.

Build finished. The HTML pages are in ../doc/html.
```

After popping back in the line 122 removed in this file (``html_static_path = ['_static']``), the build succeeded with no warnings, so it does seem to be a problem relating to ``html_static_path`` & therefore probably this which seems to be the intended replacement for setting it?

Reply:
I cannot replicate this, it might be an environment issue:

```
cd rose
pip install -e .
etc/bin/rose-make-docs clean html
```

Reply:
I'm still getting the above warnings when running those commands (though previously I ran ``make html`` directly, as I was unaware the ``make-docs`` command was still there in any form, having just looked under the ``bin/`` for any such command.)

If you still think it is an environment issue I could perhaps send some reports such as a ```conda list`` output to you, by email so as not to clutter up this Issue?

Reply:
Before I try trashing about with environments could you try upgrading your sphinx version to see if that fixes it.

Reply:
No (with ``v2.1.2``, the latest on conda)...

```console
$ etc/bin/rose-make-docs clean html
make: Entering directory `/net/home/h06/sbarth/rose.git/sphinx'
rm -rf /home/h06/sbarth/rose.git/doc/2.0.0alpha3/*
sphinx-build -b html -d /home/h06/sbarth/rose.git/doc/2.0.0alpha3/doctrees   . /home/h06/sbarth/rose.git/doc/2.0.0alpha3/html
Running Sphinx v2.1.2
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 80 source files that are out of date
updating environment: 80 added, 0 changed, 0 removed
stty: standard input: Inappropriate ioctl for device                            
reading sources... [ 58%] tutorial/cylc/runtime/configuration-consolidation/famireading sources... [ 60%] tutorial/cylc/runtime/configuration-consolidation/indereading sources... [ 61%] tutorial/cylc/runtime/configuration-consolidation/jinjreading sources... [ 62%] tutorial/cylc/runtime/configuration-consolidation/parareading sources... [ 92%] tutorial/rose/furthertopics/upgrading-macro-developmenreading sources... [100%] tutorial/rose/summary                                 
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [ 58%] tutorial/cylc/runtime/configuration-consolidation/familwriting output... [ 61%] tutorial/cylc/runtime/configuration-consolidation/jinjawriting output... [ 62%] tutorial/cylc/runtime/configuration-consolidation/paramwriting output... [100%] tutorial/rose/summary                                  
generating indices... genindex http-routingtable py-modindex
highlighting module code... [100%] rose_domain                                  
writing additional pages... search/home/h06/sbarth/miniconda3/lib/python3.7/site-packages/sphinx_rtd_theme/search.html:20: RemovedInSphinx30Warning: To modify script_files in the theme is deprecated. Please insert a <script> tag directly in your theme instead.
  {{ super() }}

copying images... [ 60%] tutorial/cylc/scheduling/../img/cylc-graph-reversible.scopying images... [ 73%] tutorial/rose/furthertopics/img/rose-macro-tutorial-appcopying images... [ 78%] tutorial/rose/furthertopics/img/rose-edit-macro-fail.pncopying images... [ 82%] tutorial/rose/furthertopics/img/rose-bunch-bush-page.pncopying images... [ 86%] tutorial/rose/furthertopics/img/rose-edit-custom-widgetcopying images... [100%] tutorial/rose/img/rosie-go-author-search.png           
copying static files... WARNING: html_static_path entry '/net/home/h06/sbarth/miniconda3/lib/python3.7/site-packages/cylc/sphinx_ext/diff_selection/_static' does not exist
WARNING: html_static_path entry '/net/home/h06/sbarth/miniconda3/lib/python3.7/site-packages/cylc/sphinx_ext/grid_table/_static' does not exist
WARNING: html_static_path entry '/net/home/h06/sbarth/miniconda3/lib/python3.7/site-packages/cylc/sphinx_ext/hieroglyph_theme_addons/_static' does not exist
WARNING: html_static_path entry '/net/home/h06/sbarth/miniconda3/lib/python3.7/site-packages/cylc/sphinx_ext/minicylc/_static' does not exist
WARNING: html_static_path entry '/net/home/h06/sbarth/miniconda3/lib/python3.7/site-packages/cylc/sphinx_ext/practical/_static' does not exist
WARNING: html_static_path entry '/net/home/h06/sbarth/miniconda3/lib/python3.7/site-packages/cylc/sphinx_ext/rtd_theme_addons/_static' does not exist
done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 6 warnings.

The HTML pages are in ../../../../../../home/h06/sbarth/rose.git/doc/2.0.0alpha3/html.

Build finished. The HTML pages are in /home/h06/sbarth/rose.git/doc/2.0.0alpha3/html.
make: Leaving directory `/net/home/h06/sbarth/rose.git/sphinx'
```

Reply:
OK, it works if the extensions are installed in "editable" mode, have put up a PR to the extensions repo.

https://github.com/cylc/cylc-sphinx-extensions/pull/7

Reply:
:tada: Yes, in conjunction with the branch on ``cylc-sphinx-extensions`` as above, the warnings disappear & the build is clean, & everything displays as it should from a quick look-see in the built pages. Great.